### PR TITLE
Query: Throw better exception message for non-composable SQL in FromSql

### LINKED ISSUE #17558: Re-introduce detection for non-composable SQL in FromSql
.Net core 3 preview 8

````csharp
  HomeCarouselViewModel viewModel = await _context.Set<HomeCarouselViewModel>().FromSqlRaw("EXEC GetHomeCarousel").AsNoTracking().FirstOrDefaultAsync(); 
````
I get this exception:
Incorrect syntax near the keyword 'EXEC'

Worked perfectly with ef core 2.2

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #17558

### CODE REVIEW

File: src/EFCore.Relational/Properties/RelationalStrings.resx [hunk 1]
```diff
@@ -494,4 +494,7 @@
   <data name="SetOperationNotWithinEntityTypeHierarchy" xml:space="preserve">
     <value>Set operations (Union, Concat, Intersect, Except) are only supported over entity types within the same type hierarchy.</value>
   </data>
-</root>
+  <data name="FromSqlNonComposable" xml:space="preserve">
+    <value>FromSqlRaw/FromSqlInterpolated API is being used with non-composable SQL but query is composing over it. Considering using `AsEnumerable` after FromSqlRaw/FromSqlInterpolated method to avoid composing on server side.</value>
```
Comment (smitpatel) [inline src/EFCore.Relational/Properties/RelationalStrings.resx]: Exception message review.
@aspnet/efteam

Comment (AndriySvyryd) [inline src/EFCore.Relational/Properties/RelationalStrings.resx]:
```suggestion
    <value>FromSqlRaw or FromSqlInterpolated was called with non-composable SQL and with a query composing over it. Considering calling `AsEnumerable` after the FromSqlRaw or FromSqlInterpolated method to perform the composition on the client side.</value>
```

File: src/EFCore.Relational/Query/QuerySqlGenerator.cs [hunk 2]
```diff
@@ -324,6 +330,15 @@ protected override Expression VisitFromSql(FromSqlExpression fromSqlExpression)
             return fromSqlExpression;
         }
 
+        private bool IsComposableSql(string sql)
+        {
+            var trimmedSql = sql.TrimStart('\r', '\n', '\t', ' ');
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Query/QuerySqlGenerator.cs]: Why not trim all whitespace?

File: src/EFCore.Relational/Query/QuerySqlGenerator.cs [hunk 3]
```diff
@@ -324,6 +330,15 @@ protected override Expression VisitFromSql(FromSqlExpression fromSqlExpression)
...
+
+            return trimmedSql.StartsWith("SELECT ", StringComparison.OrdinalIgnoreCase)
+                || trimmedSql.StartsWith("SELECT" + Environment.NewLine, StringComparison.OrdinalIgnoreCase)
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Query/QuerySqlGenerator.cs]: A compiled regexp would be more robust

File: src/EFCore.Relational/Properties/RelationalStrings.resx [hunk 4]
```diff
@@ -494,4 +494,7 @@
   <data name="SetOperationNotWithinEntityTypeHierarchy" xml:space="preserve">
     <value>Set operations (Union, Concat, Intersect, Except) are only supported over entity types within the same type hierarchy.</value>
   </data>
-</root>
+  <data name="FromSqlNonComposable" xml:space="preserve">
+    <value>FromSqlRaw or FromSqlInterpolated was called with non-composable SQL and with a query composing over it. Considering calling `AsEnumerable` after the FromSqlRaw or FromSqlInterpolated method to perform the composition on the client side.</value>
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Properties/RelationalStrings.resx]:
```suggestion
    <value>FromSqlRaw or FromSqlInterpolated was called with non-composable SQL and with a query composing over it. Consider calling `AsEnumerable` after the FromSqlRaw or FromSqlInterpolated method to perform the composition on the client side.</value>
```

Comment (smitpatel) [reply src/EFCore.Relational/Properties/RelationalStrings.resx]: I copied what you had suggested earlier. 😆

Comment (AndriySvyryd) [reply src/EFCore.Relational/Properties/RelationalStrings.resx]: I didn't spot it the first time around. I know it's hard to believe, but I am not perfect.

**Pull request merged**